lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -79,16 +84,34 @@ def transform_block_filters(self, usage_info, block_structure):
             return [block_structure.create_universal_filter()]
 
         user_groups = get_user_partition_groups(usage_info.course_key, user_partitions, user, 'id')
-        group_access_filter = block_structure.create_removal_filter(
-            lambda block_key: not (
-                has_access(user, 'staff', block_key) or
-                block_structure.get_transformer_block_field(block_key, self, 'merged_group_access').check_group_access(
-                    user_groups
+        if self.hide_access_denials:
```

Comment:
I think we can just leave this code out, since there's a transformer further down the pipe that removes anything w/ blocked access.